python/tvm/micro/contrib/zephyr.py

```diff
@@ -108,7 +108,18 @@ def __init__(
                 f"project_dir supplied to ZephyrCompiler does not exist: {project_dir}"
             )
 
+        if "qemu" in board:
```

Comment:
can you simplify: self._is_qemu = "qemu" in board

Reply:
@areusch I can but I'm actually trimming off based on that check too. If I simplify like you suggest it would be:
```
diff --git a/python/tvm/micro/contrib/zephyr.py b/python/tvm/micro/contrib/zephyr.py
index ed66ce111..e772fa610 100644
--- a/python/tvm/micro/contrib/zephyr.py
+++ b/python/tvm/micro/contrib/zephyr.py
@@ -108,15 +108,14 @@ class ZephyrCompiler(tvm.micro.Compiler):
                 f"project_dir supplied to ZephyrCompiler does not exist: {project_dir}"
             )
 
-        if "qemu" in board:
-            self._qemu = True
+        self._qemu = "qemu" in board
 
-            # For Zephyr boards that run emulated by default, i.e. "simulation: qemu" is set in
-            # the board's .yaml config file, and which their names don't have the prefix "qemu_", a
-            # suffix "-qemu" is used in microTVM to inform that the QEMU transporter has to be used.
-            # So the suffix needs to be trimmed off before the board name is passed to Zephyr.
-            if "-qemu" in board:
-                board = board.replace("-qemu", "")
+        # For Zephyr boards that run emulated by default, i.e. "simulation: qemu" is set in
+        # the board's .yaml config file, and which their names don't have the prefix "qemu_", a
+        # suffix "-qemu" is used in microTVM to inform that the QEMU transporter has to be used.
+        # So the suffix needs to be trimmed off before the board name is passed to Zephyr.
+        if "-qemu" in board:
+            board = board.replace("-qemu", "")
 
         self._board = board
```

i.e. to keep the same result I would have to check for "-qemu" anyway below in order to trim off the suffix from the board name. Am I following your suggestion correctly? IF so, I still prefer trimming off only when "qemu" is in board, so the first version.

Reply:
your diff looks right to me. I think this is one of those cases where there is not much CPU time to be saved, and in particular since this is Python, the `self._is_qemu` lookup could cost more time than the substring check. in these cases, I just favor the easiest code to read/debug, rather than adding e.g. `if self._is_qemu and "-qemu" in board` (you would the have to explain the self._is_qemu--is this intending to capture some other criteria, which may shift over time to be more than just `"qemu" in board`?).

Reply:
Over time I think that code will be updated to take care of FVP emulation too (about to be added to Zephyr, for Cortex M-55 + Ethos U-55, i.e. for MPS3 boards which are not supported by QEMU).

That said, I was actually not thinking of saving CPU cycles (I don't think there is a difference in performance between the two forms), nor about adding FVP support in the future, rather I just think that the first form is better to read/debug because it makes explicit that the trimming off applies only for a specific case where the suffix is present in the board name :) In the second form that dependency is "weak" and one maybe need to go back to the first `if` to realize that the second `if` will never be taken if the first is not taken. Anyway, since the result is the same, I'm sending a new version with the change suggested. Thanks.

Reply:
ah okay. your logic makes sense, but I guess now we have a passing PR and I think they're pretty similar. let's merge this.